### PR TITLE
[AIX] Fix for AIX build break

### PR DESCRIPTION
**Pull request by ranjitshs**
### Description
With recent changes, below build error is found under AIX. 

```
ld: 0706-012 The -p flag is not recognized.
ld: 0706-012 The -a flag is not recognized.
ld: 0706-012 The -t flag is not recognized.
ld: 0706-012 The -h flag is not recognized.
ld: 0706-012 The -= flag is not recognized.
ld: 0706-012 The -$ flag is not recognized.
ld: 0706-012 The -$ flag is not recognized.
ld: 0706-012 The -O flag is not recognized.
ld: 0706-027 The -R IGIN flag is ignored.

collect2: error: ld returned 255 exit status
```

### Motivation and Context
AIX linker doesn't support -rpath option , so blocking this option under AIX.

### CODE REVIEW

**Comment by ranjitshs** [comment]
@snnn @tianleiwu 
Small fix for AIX build break. Please review and trigger workflows.

**Review by snnn — DISMISSED** [review]

File: cmake/onnxruntime.cmake [hunk 1]
```diff
@@ -122,7 +122,7 @@ else()
   else()
     onnxruntime_add_shared_library(onnxruntime ${CMAKE_CURRENT_BINARY_DIR}/generated_source.c )
   endif()
-  if(NOT APPLE)
+  if(NOT APPLE AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "AIX")
```
Comment (snnn) [inline cmake/onnxruntime.cmake:125]: Can we use https://cmake.org/cmake/help/latest/module/CheckLinkerFlag.html instead? That would be more general.

Comment (snnn) [reply cmake/onnxruntime.cmake:125]: Or detect https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_LINKER_ID.html ?

Comment (ranjitshs) [reply cmake/onnxruntime.cmake]: @snnn
made the suggested changes and using  check_linker_flag for checking linker support for path

**Comment by ranjitshs** [comment]
@snnn 
in this PR, i am fixing one typo mistake. 
Please trigger workflows

**Review by snnn — APPROVED** [review]

**Pull request merged**